Unify Event Filtering Public Beta [DOC-1107] (#7602)
segmentio/segment-docs: src/unify/event-filtering.md

---
title: Unify Event Filtering
plan: unify
---

Unify Event Filtering lets you control which events reach your Unify space. 

> info "Public Beta"
> Unify Event Filtering is in public beta, and Segment is actively working on this feature. Some functionality may change before it becomes generally available. 

## Overview

> With Unify Event Filtering, you can use block events that aren’t useful for [Identity Resolution](/docs/unify/identity-resolution/) or downstream activation, keeping your space cleaner and more efficient.

Reviewer: ```suggestion
With Unify Event Filtering, you can use block events that aren’t useful for [Identity Resolution](/docs/unify/identity-resolution/) or downstream activation, which keeps your space cleaner and more efficient.
```

| Setup            | Created through API, visible in Unify UI                         | Created in UI or API                    |
| Timing           | Before Identity Resolution                                       | Before events get sent to a destination |

This helps simplify your setup and reduce noise in profile data without needing repeaters, webhooks, or custom logic.

## How filtering works 

Unify Event Filtering evaluates every incoming event before it enters your Unify space. If the event matches any of your filters, it gets dropped.

Keep the following in mind as you work with Unify Event Filtering:

- Filters are configured at the space level. You don’t create filters per source. Instead, one set of filters applies to all sources connected to the space.
- The filtering logic is based on [Filter Query Language](/docs/api/public-api/fql/). If you’ve used Destination Filters before, it works the same way. You can write expressions to match events based on type, name, traits, properties, or values.
- Matching is inclusive: if an event matches any filter, Segment drops it.

> - Filtering happens **before** Identity Resolution; dropped events never reach the profile graph and won’t affect trait updates or [computed traits](/docs/unify/traits/computed-traits/).

Reviewer: ```suggestion
- Filtering happens **before** Identity Resolution: dropped events never reach the profile graph and won’t affect trait updates or [computed traits](/docs/unify/traits/computed-traits/).
```

As a result, Unify Event Filtering helps you keep profile data clean from the start, without having to preprocess or transform events outside of Segment.

## Creating and managing filters

> During public beta, you'll use the [Space Filter API](https://docs.segmentapis.com/tag/Space-Filters/){:target="_blank"} to create and manage all Unify event filters. The API lets you define filters using FQL, name them, enable or disable them, and delete them. Each Unify space can have up to 10 filters. Any event that matches one of those filters gets dropped before it reaches Unify.

Reviewer: ```suggestion
During public beta, you'll use the [Space Filter API](https://docs.segmentapis.com/tag/Space-Filters/){:target="_blank"} to create and manage all Unify event filters. The API lets you define filters using FQL, name them, enable or disable them, and delete them. Each Unify space can have up to 10 filters. Any event that matches 1 of those filters gets dropped before it reaches Unify.
```

## Best practices

Unify Event Filtering is most useful when you want to keep noisy, irrelevant, or duplicate data out of your Unify space. The following table lists best practices to help you get the most value out of filtering:

| Tip                      | Why it matters                                                                                                                                                                       |
| ------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| Filter early             | Prevents irrelevant events from affecting profile data or identity resolution.                                                                                                       |
| Drop obvious noise       | Start with telemetry, test data, or internal events.                                                                                                                                 |

> | Keep it simple           | A few targeted filters are easier to manage than multiple, complex filters.                                                                                                          |

Reviewer: ```suggestion
| Keep it simple           | A few targeted filters are easier to manage than multiple complex filters.                                                                                                          |
```
